middleware/gating.go

```diff
@@ -0,0 +1,70 @@
+package middleware
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/livepeer/catalyst-api/config"
+	catErrs "github.com/livepeer/catalyst-api/errors"
+	"github.com/livepeer/catalyst-api/log"
+	"github.com/livepeer/catalyst-api/playback"
+	accesscontrol "github.com/livepeer/catalyst/cmd/catalyst-node/handlers/access-control"
+)
+
+type playbackAccessControlRequest struct {
+	Type      string `json:"type"`
+	AccessKey string `json:"accessKey"`
+	Stream    string `json:"stream"`
+}
+
+type GatingHandler struct {
+	AccessControl *accesscontrol.PlaybackAccessControl
+}
+
+func NewGatingHandler(gateURL string) *GatingHandler {
+	return &GatingHandler{
+		AccessControl: accesscontrol.NewPlaybackAccessControl(gateURL),
+	}
+}
+
+func (h *GatingHandler) GatingCheck(next httprouter.Handle) httprouter.Handle {
+	return func(w http.ResponseWriter, req *http.Request, params httprouter.Params) {
+		var requestID = config.RandomTrailer(8)
+		req.Header.Set("requestID", requestID)
```

Comment:
Perhaps a common helper like `getOrSetRequestId(req.Header)` would be useful across middlewares/handlers.